### DISCUSSION TITLE
Upgradation of openssl version from 1.0.2 to 3.3.2 in yocto (poky) in IMX6 processor

### DISCUSSION THREAD

**Discussion by abineshkumarcoherent** [openssl/openssl#25634] [Q&A]
We are in the process of upgrading OpenSSL from version 1.0.2 to 3.3.2 in our Yocto environment. During the build, we encountered an issue related to the configuration step. The specific error log is as follows:

| DEBUG: Executing shell function do_configure
| Can't open perl script "perlpath.pl": No such file or directory
| WARNING: exit code 2 from a shell command.
| ERROR: Function failed: do_configure

It appears that the perlpath.pl script is missing, and we have not been able to identify a patch or solution to address this. Could you kindly provide guidance on how to resolve this issue?

**Comment by nhorman** [comment]
perlpath.pl was originally in util/perlpath.pl, but was removed in 843666ffdc0745e1aed5d665ec18d57263e32079.  the script just rewrites some perl scripts which had the line:
```
#!/usr/local/bin/perl
```
to a real path on the local system

That mutation isn't required in our build scripts any longer, so the script isn't needed, and can just be removed from the spec file (working under the assumption that you have a spec file from the 1.0.2 era

**Reply by mattcaswell** [reply]
> Executing shell function do_configure

Looks like wherever you define that shell function could do with some update.

**Comment by abineshkumarcoherent** [comment]
Is there is any steps or documents  to update openssl to the latest version  in imx6 Dual light  boards
